src/PMGDQuery.cc

```diff
@@ -209,7 +209,20 @@ void PMGDQuery::set_property(PMGDProp* p, const std::string& key,
             break;
 
         default:
-            p->set_type(PMGDProp::NoValueType);
+            // For some reason, Jsoncpp is not parsing a long integer
+            // and using largest int as mentioned in their specification.
+            // So large integers come here. Handle that for now and throw
+            // an exception when we don't know what is going on so we can
+            // catch it instead of failing silently.
+            if (val.isInt64()) {
+                p->set_type(PMGDProp::IntegerType);
+                p->set_int_value(val.asInt64());
+            }
+            else {
+                printf("%s\n", key.c_str());
+                throw ExceptionCommand(PMGDTransactiontError,
+                                       "Object Type Error");
+            }
```

Comment:
Actually that function intValue() is technically supposed to recognize large ints. I am not sure why it is not. I looked up some discussion online. Anyway, that's why I put it here.

Reply:
One thing i am not sure is, if it starts working for some version of jsoncpp library, if I should put asInt64() in the case of intValue. I didn't do it this time cause most of the calls to this function are for getting a ref value which is actually int and not int64.

Reply:
I suggest putting the exact code that is currently in the default case into a uintValue case. The isInt64 check still needs to be there to ensure that it isn't out of range.  It can be combined with the intValue case: when ValueType is intValue, IsInt64 is a no-op (always returns true).

The default case can just throw.